### PR TITLE
Python 3.6 compatibility

### PR DESCRIPTION
**Pull request by wkitka**
Make dsconfig compatible with Python 3.6 and drop support for Python 2.
Tested features: xls2json, json2tango, dsconfig.dump, dsconfig.viewer.

### CODE REVIEW

**Comment by ajoubertza** [comment]
@wkitka Thanks - we (SKA) were planning to do this long ago, but never got time to finish it.

We did add some tests as we wanted to increase the test coverage before migrating.  Those are in a mirrored repo:  https://gitlab.com/ska-telescope/lib-maxiv-dsconfig/-/tree/SAR-2/expand_test_coverage.   You may want to include those, or at least try them out against your new code.

Note that some of the files are no longer needed - there is an open PR removing them: #12.

Would you mind filling in the description for this PR?  For example, is this update dropping support for Python 2, or is it now compatible with both 2.7 and 3.6?  Any limitations?  Etc.

File: setup.py [hunk 1]
```diff
@@ -5,20 +5,22 @@
 setup(
     # Package
     name="python-dsconfig",
-    version="1.4.0",
-    packages=['dsconfig', 'dsconfig.appending_dict'],
+    version="1.5.0",
+    packages=["dsconfig", "dsconfig.appending_dict"],
     description="Library and utilities for Tango device configuration.",
     # Requirements
```
Comment (ajoubertza) [inline setup.py:11]: Adding a `python_requires` line would be good.
For example: 
https://github.com/pypa/sampleproject/blob/150be5fc818484b78816529bc9ec59440a9bcdab/setup.py#L137

**Review by ajoubertza — COMMENTED** [review]
I only looked at `setup.py`.

File: setup.py (hunk 1 above)
Comment (wkitka) [reply setup.py:11]: Thanks for comments! I'll improve this PR.

Comment (wkitka) [reply setup.py:11]: I included both mentioned PRs and added description

**Comment by bamartos** [comment]
Hej, 

It looks also good from my side, shall we proceed to the merge?

**Comment by wkitka** [comment]
> Hej,
> 
> It looks also good from my side, shall we proceed to the merge?

Hej @bamartos 
I think so. Thanks for checking!

**Review by ajoubertza — APPROVED** [review]
Thanks @wkitka 
Fingers crossed (since this repo has no CI!) 🙂

**Comment by wkitka** [comment]
@bamartos 
would you do the honors and merge it?

**Pull request merged**